controllers/heatengine_controller.go

```diff
@@ -528,6 +515,38 @@ func (r *HeatEngineReconciler) reconcileUpgrade(ctx context.Context, instance *h
 	return ctrl.Result{}, nil
 }
 
+// getSecret - get the specified secret, and add its hash to envVars
+func (r *HeatEngineReconciler) getSecret(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *heatv1beta1.HeatEngine,
```

Comment:
I'm not a huge fan of this being duplicated for each controller. I believe the instance here would satisfy the `client.Object` interface, which would mean we can create a single function that isn't a method of the reconciler, where the instance passed in is of type `client.Object`. 

But we can merge this now and then address that point in a follow up PR. I assume all of the other PR's are doing the same thing as defined here. So for consistency sake, it's best to just merge and we'll look at improving the implementation later.

Reply:
Ah, we get Status. That isn't a client.Object field. But we still might be able to find a common type that satisfies this requirement and enables us to make this code more generic. I'll dig into it and see if I can propose something that we can collab on.

Reply:
I still have three more of these PRs to do, so far every single one has had some new thing I've had to learn so we'll see how barbican / manila / telemetry go